test/widgets/content_test.dart

```diff
@@ -290,6 +339,88 @@ void main() {
 
   testContentSmoke(ContentExample.mathBlock);
 
+  testContentSmoke(ContentExample.strong);
+
+  testContentSmoke(ContentExample.emphasis);
+
+  group('InlineCodeNode', () {
+    testContentSmoke(ContentExample.inlineCode);
+
+    testWidgets('maintains font-size ratio with surrounding text', (tester) async {
+      await prepareContentBare(tester,
+        '<h1>header-plain<code>header-code</code></h1>\n'
+        '<p>paragraph-plain<code>paragraph-code</code></p>');
+
+      final headerRootSpan = tester.renderObject<RenderParagraph>(find.textContaining('header')).text;
+      final headerPlainStyle = mergedStyleOfSubstring(headerRootSpan, 'header-plain');
+      final headerCodeStyle = mergedStyleOfSubstring(headerRootSpan, 'header-code');
+
+      final paragraphRootSpan = tester.renderObject<RenderParagraph>(find.textContaining('paragraph')).text;
+      final paragraphPlainStyle = mergedStyleOfSubstring(paragraphRootSpan, 'paragraph-plain');
+      final paragraphCodeStyle = mergedStyleOfSubstring(paragraphRootSpan, 'paragraph-code');
+
+      check(headerPlainStyle).isNotNull().fontSize.isNotNull();
+      check(headerCodeStyle).isNotNull().fontSize.isNotNull();
+      check(paragraphPlainStyle).isNotNull().fontSize.isNotNull();
+      check(paragraphCodeStyle).isNotNull().fontSize.isNotNull();
+
+      final ratioInHeader = headerCodeStyle!.fontSize! / headerPlainStyle!.fontSize!;
+      final ratioInParagraph = paragraphCodeStyle!.fontSize! / paragraphPlainStyle!.fontSize!;
+
+      check(ratioInHeader).isLessThan(1);
+      check(ratioInParagraph).isLessThan(1);
+
+      // Empirically we might have e.g. 0.825 and 0.8250000000000001.
+      check((ratioInHeader - ratioInParagraph).abs()).isLessThan(0.001);
+
+      check(headerPlainStyle.fontSize! / paragraphPlainStyle.fontSize!)
+        .isGreaterOrEqual(1.1);
+    });
+  });
+
+  group('UserMentionNode', () {
+    testContentSmoke(ContentExample.userMentionPlain);
+    testContentSmoke(ContentExample.userMentionSilent);
+    testContentSmoke(ContentExample.groupMentionPlain);
+    testContentSmoke(ContentExample.groupMentionSilent);
+
+    testWidgets('maintains font-size ratio with surrounding text', (tester) async {
+      await prepareContentBare(tester,
+        '<h1>'
+          'header-plain '
+          '<span class="user-mention" data-user-id="13313">@Chris Bobbe</span>'
+        '</h1>\n'
+        '<p>'
+          'paragraph-plain '
+          '<span class="user-mention" data-user-id="13313">@Chris Bobbe</span>'
+        '</p>');
+
+      final headerRootSpan = tester.renderObject<RenderParagraph>(find.textContaining('header')).text;
+      final headerPlainStyle = mergedStyleOfSubstring(headerRootSpan, 'header-plain ');
+      final headerMentionStyle = mergeSpanStylesOuterToInner(headerRootSpan,
+        (span) => span is WidgetSpan && span.child is UserMention);
```

Comment:
This test seems not to actually work — if I check out the old code in `lib`, the test still passes.

Reply:
There's ultimately a text span that has the mentioned user's name somewhere inside that `UserMention` widget, and the text style will live there, right?

Specifically I think if we find the `RenderParagraph` that text span is part of, and merge together its styles in the same way as before, then that should tell us the effective style for that span. In particular I believe all the defaulting and merging that happens outside of `RenderParagraph` and SkParagraph winds up going into the `TextStyle` that gets placed at the root of that `RenderParagraph`'s span tree.